### DISCUSSION TITLE
Denys all connections when running in docker

### DISCUSSION THREAD

**Discussion by tacyarg** [uNetworking/uWebSockets.js#219] [Q&A] [answered]
If you listen using "localhost" in **docker** the app will refuse all inbound connections.

```js
BEFORE:
app.listen('localhost', 2002, () => {
    console.log("FUCK")
})

AFTER:
app.listen('*', 2002, () => {
    console.log("FUCK")
})
```

Just thought i'd save someone else an entire day debugging this torturous problem.
Now the docker container works properly.

**Comment by ghost** [comment, marked as answer]
You can also just do app.listen(2002). If you want to bind to any one interface you have to make sure you actually have that interface up, using something like ifconfig

**Comment by lagden** [comment]
@tacyarg 

I had the same issue!  
I just did:

```js
const PORT = 3000
app.listen('::', PORT, token => {
  debug.info(token ? `Server listening on port ${PORT}` : `Failed to listen to port ${PORT}`)
})
```

Now, the container is listening the  `IP` and `PORT` correctly...

```shell
~/app $ netstat -plnt
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.11:46767        0.0.0.0:*               LISTEN      -
tcp        0      0 :::3000                 :::*                    LISTEN      1/node
```

PS: Build from `node:12.14-alpine`
